x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/model_select_option.tsx

```diff
@@ -5,56 +5,203 @@
+      {i18n.translate(
+        'xpack.enterpriseSearch.content.indices.pipelines.modelSelectOption.startButton.label',
+        {
+          defaultMessage: 'Start',
+        }
+      )}
+    </EuiButton>
+  );
+};
+
+export const ModelMenuPopover: React.FC<{
+  onClick: () => void;
+  closePopover: () => void;
+  isOpen: boolean;
+  modelDetailsPageUrl?: string;
+}> = ({ onClick, closePopover, isOpen, modelDetailsPageUrl }) => {
+  return (
+    <EuiPopover
+      button={<EuiButtonIcon onClick={onClick} iconType="boxesHorizontal" />}
+      isOpen={isOpen}
+      closePopover={closePopover}
+      anchorPosition="leftCenter"
+      panelPaddingSize="none"
+    >
+      <EuiContextMenu panels={getContextMenuPanel(modelDetailsPageUrl)} initialPanelId={0} />
+    </EuiPopover>
+  );
+};
+
+export const ModelSelectOption: React.FC<ModelSelectOptionProps> = ({
+  modelId,
+  title,
+  description,
+  license,
+  deploymentState,
+  deploymentStateReason,
+  modelDetailsPageUrl,
+  isPlaceholder,
+  checked,
+}) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const onMenuButtonClick = () => setIsPopoverOpen((isOpen) => !isOpen);
+  const closePopover = () => setIsPopoverOpen(false);
+
+  const { createModel, startModel } = useActions(ModelSelectLogic);
+  const { areActionButtonsDisabled } = useValues(ModelSelectLogic);
+
   return (
-    <EuiFlexGroup direction="column" gutterSize="xs">
-      <EuiFlexItem>
-        <EuiTitle size="xs">
-          <h4>{title ?? model.model_id}</h4>
-        </EuiTitle>
+    <EuiFlexGroup alignItems="center">
+      {/* Selection radio button */}
+      <EuiFlexItem grow={false} style={{ flexShrink: 0 }}>
+        <EuiRadio id={modelId} checked={checked === 'on'} onChange={() => null} />
       </EuiFlexItem>
-      <EuiFlexItem>
-        <EuiFlexGroup gutterSize="s" alignItems="center" justifyContent="flexEnd">
-          {title && (
-            <EuiFlexItem>
-              <EuiTextColor color="subdued">{model.model_id}</EuiTextColor>
-            </EuiFlexItem>
-          )}
-          <EuiFlexItem grow={false}>
-            <TrainedModelHealth
-              modelState={parseModelStateFromStats(model)}
-              modelStateReason={parseModelStateReasonFromStats(model)}
-            />
+      {/* Title, model ID, description, license */}
+      <EuiFlexItem style={{ overflow: 'hidden' }}>
+        <EuiFlexGroup direction="column" gutterSize="xs">
+          <EuiFlexItem>
+            <EuiTitle size="xs">
+              <h4>{title}</h4>
+            </EuiTitle>
           </EuiFlexItem>
-          <EuiFlexItem grow={false}>
-            <EuiFlexGroup gutterSize="xs">
-              <EuiFlexItem>
-                <MLModelTypeBadge type={type} />
-              </EuiFlexItem>
+          <EuiFlexItem>
+            <EuiTextColor color="subdued">{modelId}</EuiTextColor>
+          </EuiFlexItem>
+          <EuiFlexItem>
+            <EuiFlexGroup gutterSize="xs" alignItems="center">
```

Comment:
Is it possible to have neither a license or description?

Reply:
Yes it's possible. Should I wrap the parent `EuiFlexItem` in `{(license || description) && ... }`?

Reply:
Yes I think so.

Reply:
Done.